posixlib/src/main/scala/scala/scalanative/posix/time.scala

```diff
@@ -57,23 +67,27 @@ object timeOps {
   }
 
   implicit class tmOps(val ptr: Ptr[tm]) extends AnyVal {
-    def tm_sec: CInt              = ptr._1
-    def tm_min: CInt              = ptr._2
-    def tm_hour: CInt             = ptr._3
-    def tm_mday: CInt             = ptr._4
-    def tm_mon: CInt              = ptr._5
-    def tm_year: CInt             = ptr._6
-    def tm_wday: CInt             = ptr._7
-    def tm_yday: CInt             = ptr._8
-    def tm_isdst: CInt            = ptr._9
-    def tm_sec_=(v: CInt): Unit   = ptr._1 = v
-    def tm_min_=(v: CInt): Unit   = ptr._2 = v
-    def tm_hour_=(v: CInt): Unit  = ptr._3 = v
-    def tm_mday_=(v: CInt): Unit  = ptr._4 = v
-    def tm_mon_=(v: CInt): Unit   = ptr._5 = v
-    def tm_year_=(v: CInt): Unit  = ptr._6 = v
-    def tm_wday_=(v: CInt): Unit  = ptr._7 = v
-    def tm_yday_=(v: CInt): Unit  = ptr._8 = v
-    def tm_isdst_=(v: CInt): Unit = ptr._9 = v
+    def tm_sec: CInt                = ptr._1
+    def tm_min: CInt                = ptr._2
+    def tm_hour: CInt               = ptr._3
+    def tm_mday: CInt               = ptr._4
+    def tm_mon: CInt                = ptr._5
+    def tm_year: CInt               = ptr._6
+    def tm_wday: CInt               = ptr._7
+    def tm_yday: CInt               = ptr._8
+    def tm_isdst: CInt              = ptr._9
+    def tm_gmtoff: CLong            = ptr._10
```

Comment:
What to do here when handling "short" 36 byte 
`struct tm` , and below in the "set" code is a good question.
Should probably be an UnsupportedOperation (spelling?) 
Exception, alternative is to silently do nothing.

Reply:
Let ask @sjrd. For me we should broke compatibility with short structure because on modern system using short cause memory corruption.

Yes, it was a case in my application from real life that I hunted for a couple of months.

Reply:
I believe that posixlib needs to continue to work with the basic POSIX short structure. 
I think they should not be abandoned.   Much as I would like to be over and done
with 32 bit systems, many contemporary and modern systems are and will be
32 bit. (someday talk to me about joy of file sizes).

PR #2066 fixes the memory corruption. 
It, and its predecessor do not offer access to the extended POSIX XSI fields.

I believe that with a few surgical changes, like not writing into fields that do not exist
on machines which use short structures, offers an improvement.

I do not want to be an impediment to progress, but I also do not want to see 
one memory corruption swapped for another.

Would it be more useful if I did a Work In Progress PR, based on this one, to illustrate
the few changes I am trying to describe?  This whole topic gets pretty complex pretty
quickly.  tzset() not being called at the start of execution is a real pain.